File: model/product.go
package model

type Product struct {
	HotelId       int64   `json:"hotel,omitempty"`
	Id            int64   `json:"id,omitempty"`
	UserId        int64   `json:"usuario,omitempty"`
	GroupId       int64   `json:"grupo,omitempty"`
	CfpopId       int64   `json:"cfop,omitempty"`
	Code          string  `json:"codigo,omitempty"`
	BarCode       string  `json:"codigo_barras,omitempty"`
	Description   string  `json:"descricao,omitempty"`
	Comment       string  `json:"observacao,omitempty"`
	MinStock      float64 `json:"estoque_minimo,omitempty"`
	MaxStock      float64 `json:"estoque_maximo,omitempty"`
	Stock         float64 `json:"estoque,omitempty"`
	CreatedAt     string  `json:"data_cadastro,omitempty"`
	Cost          float64 `json:"custo,omitempty"`
	Price         float64 `json:"preco,omitempty"`
	ProfitMargin  float64 `json:"margem_lucro,omitempty"`
	IsActive      string  `json:"activo,omitempty"`
	StockWatch    string  `json:"controla_estoque,omitempty"`
	Buyings       float64 `json:"total_compras,omitempty"`
	Sales         float64 `json:"total_vendas,omitempty"`
	Consumption   float64 `json:"total_consumo,omitempty"`
	Unit          string  `json:"unidade,omitempty"`
	Outlet        float64 `json:"total_saida,omitempty"`
	IcmsEcf       string  `json:"icms_ecf,omitempty"`
	Tax           float64 `json:"imposto,omitempty"`
	FederalTax    float64 `json:"imposto_federal,omitempty"`
	StateTax      float64 `json:"imposto_estadual,omitempty"`
	MunicipalTax  float64 `json:"imposto_municipal,omitempty"`
	TaxSource     float64 `json:"fonte_imposto,omitempty"`
	PreparingTime int64   `json:"tempo_preparo,omitempty"`
	Discrete      string  `json:"descritivo,omitempty"`
	ServiceTax    float64 `json:"taxa_servico,omitempty"`
	Alcohool      string  `json:"bebida_alcoolica,omitempty "`
	Spotlight     string  `json:"destaque,omitempty"`
}

File: frontend/lib/models/product.dart
import 'package:datacaixa/common/contants.dart' as c;

class Product {
  int identifier;
  int hotelId;
  int productId;
  int userId;
  int productGroupId;
  int cfopId;
  String code;
  String description;
  String comment;
  double minStock;
  double maxStock;
  double stock;
  String createdAt;
  double cost;
  double price;
  double profitMargin;
  String isActive;
  String stockWatch;
  double buyings;
  double sales;
  double consumption;
  String unit;
  double outlet;
  String icmsEcf;
  double tax;
  double federalTax;
  double stateTax;
  double municipalTax;
  double taxSource;
  int preparingTime;
  String discrete;
  double serviceTax;
  String alcohol;
  String spotlight;

  Product();

  Product.add({this.productId, this.productGroupId, this.description, this.createdAt,
      this.price, this.sales});

  Map<String, dynamic> toMap() {
    var map = <String, dynamic>{
      c.productId: productId,
      c.productGroupId: productGroupId,
      c.description: description,
      c.createdAt: createdAt,
      c.price: price,
      c.sales: sales,
    };
    if(identifier != null){
      map[c.identifier] = identifier;
    }
    return map;
  }

  Product.fromMap(Map<String, dynamic> map){
    identifier = map[c.identifier];
    productId = map[c.productId];
    productGroupId = map[c.productGroupId];
    description = map[c.description];
    createdAt = map[c.createdAt];
    price = map[c.price];
    sales = map[c.sales];
  }

  factory Product.fromJson(Map<String, dynamic> json) => Product.add(
    productId: json["id"],
    productGroupId: json["grupo"],
    description: json["descricao"],
    createdAt: json["data_cadastro"],
    price: json["preco"].toDouble(),
    sales: json["total_vendas"] == null ? null : json["total_vendas"].toDouble(),
  );

  Map<String, dynamic> toJson() => {
    "id": productId,
    "grupo": productGroupId,
    "descricao": description,
    "data_cadastro": createdAt,
    "preco": price,
    "total_vendas": sales == null ? null : sales,
  };

  @override
  String toString() {
    return 'Product{identifier: $identifier, productId: $productId, productGroupId: $productGroupId, description: $description, createdAt: $createdAt, price: $price, sales: $sales}';
  }
}